src/common/components/Auth/SignupForm.tsx

```diff
@@ -0,0 +1,125 @@
+"use client";
+
+import { useState } from "react";
+import { useForm, type SubmitHandler } from "react-hook-form";
+
+import { Input } from "@/common/components/Input";
+import { Button } from "@/common/components/Button";
+import { LockIcon } from "@/common/components/CustomIcon/LockIcon";
+import { EyeIcon } from "@/common/components/CustomIcon/EyeIcon";
+import { EyeSlashIcon } from "@/common/components/CustomIcon/EyeSlashIcon";
+import { EnvelopeIcon } from "@/common/components/CustomIcon/EnvelopeIcon";
+
+type SignupFormInputs = {
+  email: string;
+  password: string;
+  confirmPassword: string;
+};
+
+export const SignupForm = () => {
+  const [isPwdVisible, setIsPwdVisible] = useState(false);
+  const [isCfmPwdVisible, setIsCfmPwdVisible] = useState(false);
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors, isSubmitting },
+  } = useForm<SignupFormInputs>({ mode: "onTouched" });
+  const onSubmit: SubmitHandler<SignupFormInputs> = async (data) => {
+    if (isSubmitting) return;
+
+    // TODO: replace with create account logic
+    console.log(data);
+    await new Promise((r) => setTimeout(r, 2000));
+    !!errors.email || !!errors.password || !!errors.confirmPassword
+      ? alert("error creating account!")
+      : alert("account created!");
+
+    reset({ password: "", confirmPassword: "" });
+  };
+
+  const isValidEmail = (email: string) =>
+    // TODO: replace with supported school email validation logic
+    /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email) && email.endsWith("smu.edu.sg");
+
+  return (
+    <form
+      className="flex w-full flex-col gap-6"
+      onSubmit={handleSubmit(onSubmit)}
+    >
+      <Input
+        label="School Email Address"
+        leftContent={<EnvelopeIcon size={24} />}
+        placeholder="john.doe.2023@smu.edu.sg"
+        isError={!!errors.email}
+        helperText={errors.email?.message}
+        autoComplete="on"
+        registerFormProps={register("email", {
+          required: "Please enter a valid school email address",
+          validate: (email) =>
+            isValidEmail(email) || "Please enter a valid school email address",
+        })}

```

Comment:
I think instead of modifying `Input`, you can just simply spread the register method directly here IMO. I think it'll make `Input` more form agnostic

```suggestion
       {...register("email", {
          required: "Please enter a valid school email address",
          validate: (email) =>
            isValidEmail(email) || "Please enter a valid school email address",
        })}

```

Reply:
somehow just using object spread on register doesn't work with our custom input component, probably due to some props not being properly passed onto the native input component

when i use obj spread on `register` per suggested, the form input is no longer registered:

![image](https://github.com/AfterClass-io/afterclass.io-v2/assets/13061926/af80f1c6-280f-4f14-8e7e-e5fad89b1d24)

vs this workaround:

![image](https://github.com/AfterClass-io/afterclass.io-v2/assets/13061926/3c1b71a6-c136-4373-acec-aaad0877687d)

Reply:
added zod with react-hook-forms at 09e7ff2!

Reply:
I added the fix here https://github.com/AfterClass-io/afterclass.io-v2/pull/38/commits/f4080aa60e54155540ec39f0ea618a35921c3d38. Since I did it before you did zod, I think there's some merging issue with using zod vs no zod. Do help to resolve!

Reply:
fixed at cb25328!